okapi-core/src/test/java/org/folio/okapi/service/impl/DockerModuleHandleTest.java

```diff
@@ -136,6 +151,68 @@ private void dockerMockHandle(RoutingContext ctx) {
     }
   }
 
+  boolean pullImage(TestContext context, Vertx vertx, String host, JsonObject conf) {
+    LaunchDescriptor ld = new LaunchDescriptor();
+    ld.setWaitIterations(2);
+    ld.setDockerImage("folioci/mod-x");
+    ld.setDockerPull(true);
+    Ports ports = new Ports(9232, 9233);
+    DockerModuleHandle dh = new DockerModuleHandle(vertx, ld,
+        "mod-users-5.0.0-SNAPSHOT", ports, host,
+        9231, conf);
+    Async async = context.async();
+    AtomicBoolean succeeded = new AtomicBoolean();
+    dh.pullImage().onComplete(done -> {
+      succeeded.set(done.succeeded());
+      async.complete();
+    });
+    async.await();
+    return succeeded.get();
+  }
+
+  @Test
+  public void testDockerPull(TestContext context) {
+    Vertx vertx = Vertx.vertx();
+    int dockerPort = 9231;
+
+    Router router = Router.router(vertx);
+    router.routeWithRegex("/.*").handler(this::dockerMockHandle);
+
+    HttpServerOptions so = new HttpServerOptions().setHandle100ContinueAutomatically(true);
+    HttpServer listen = vertx.createHttpServer(so)
+        .requestHandler(router)
+        .listen(dockerPort, context.asyncAssertSuccess());
+    dockerPullJson = new JsonObject().put("message", "some message");
+    dockerPullStatus = 200;
+
+    JsonObject conf = new JsonObject().put("dockerUrl", "tcp://localhost:" + dockerPort);
+    context.assertTrue(pullImage(context, vertx, "localhost", conf));
+
+    conf.put("dockerRegistries", new JsonArray());
+    context.assertFalse(pullImage(context, vertx, "localhost", conf));
+
+    conf.put("dockerRegistries", new JsonArray()
+        .addNull()
+        .add(new JsonObject().put("username", "x").put("password", "y")));
+    context.assertFalse(pullImage(context, vertx, "localhost", conf));
+
+    conf.put("dockerRegistries", new JsonArray()
+        .add(new JsonObject().put("username", "x").put("password", "y"))
+        .add(new JsonObject().put("username", "x").put("password", "x"))
+        .add(new JsonObject().put("username", "x").put("password", "z")));
+    context.assertTrue(pullImage(context, vertx, "localhost", conf));
+
+    conf.put("dockerRegistries", new JsonArray()
+        .add(new JsonObject().put("registry", "localhost:5000")));
+    context.assertTrue(pullImage(context, vertx, "localhost", conf));
+
+    conf.put("dockerRegistries", new JsonArray()
+        .add(new JsonObject().put("registry", "localhost:5000/")));
+    context.assertTrue(pullImage(context, vertx, "localhost", conf));
```

Comment:
The host value was "localhost:5000" before the last commit.
If we always use "localhost" this parameter can be removed and it can be hardcoded in pullImage.
How can we unit test that the "registry" parameter is handled correctly?

Reply:
There's one localhost:5000 and localhost:5000/ now.

Reply:
Yes. That "localhost" is the containerHost.. something completely different. And that can be hard-coded.